modules/api/src/main/java/org/eclipse/xpanse/api/AdminServicesApi.java

```diff
@@ -46,12 +48,27 @@
 @Secured({ROLE_ADMIN})
 public class AdminServicesApi {
 
-    @Resource
-    private IdentityProviderManager identityProviderManager;
+    private final IdentityProviderManager identityProviderManager;
 
     @Value("${spring.datasource.url:jdbc:h2:file:./testdb}")
     private String dataSourceUrl;
 
+    private final TerraformApi terraformApi;
+
+    private final ApiClient apiClient;
+
+    /**
+     * The constructor.
+     */
+    public AdminServicesApi(IdentityProviderManager identityProviderManager,
+                            TerraformApi terraformApi, ApiClient apiClient,
+                            @Value("${apiClient.basePath:}") String baseUrl) {
```

Comment:
@jinyangyang222 we must not update the basepath here. Instead we need to update the bean directly at startup and simply use it everywhere, 
example - https://github.com/Zomzog/openapi-spring-client-token/blob/master/front/src/main/java/bzh/zomzog/demo/openapispringclienttoken/front/config/BackendClientConfig.java

Reply:
ok,fixed